adaboost/cuda/core/operations_impl.cu

```diff
@@ -257,9 +257,106 @@ namespace adaboost
                  result.get_cols(),
 				 result.get_rows());
             }
+            
+            template <class data_type_vec, class data_type_ret>
+            __global__ 
+            void find_maximum_kernel(
+            data_type_ret (*func_ptr)(data_type_vec),
+            data_type_vec *vec, 
+            data_type_vec *max, 
+            int *mutex, 
+            unsigned int size)
+            {
+                // unsigned int index = threadIdx.x + blockIdx.x*blockDim.x;
+                // unsigned int stride = gridDim.x*blockDim.x;
+                // unsigned int offset = 0;
+            
+                // __shared__ float cache[MAX_BLOCK_SIZE];
+            
+            
+                // float temp = func_ptr(vec[index + offset]);
+                // while(index + offset < size){
+                //     temp = fmaxf(temp, func_ptr(vec[index + offset]));
+            
+                //     offset += stride;
+                // }
+            
+                // cache[threadIdx.x] = temp;
+            
+                // __syncthreads();
+            
+            
+                // // reduction
+                // unsigned int i = blockDim.x/2;
+                // while(i != 0){
+                //     if(threadIdx.x < i){
+                //         cache[threadIdx.x] = fmaxf(cache[threadIdx.x], cache[threadIdx.x + i]);
+                //     }
+            
+                //     __syncthreads();
+                //     i /= 2;
+                // }
+            
+                // if(threadIdx.x == 0){
+                //     while(atomicCAS(mutex,0,1) != 0);  //lock
+                //     *max = fmaxf(*max, cache[0]);
+                //     atomicExch(mutex, 0);  //unlock
+                // }
+                * max = func_ptr(1);
+                // * max = 2;
+            }
+
+            // template <class data_type_vec, class data_type_ret>
+            // using func_t = data_type_ret (*)(data_type_vec);
```

Comment:
Is the commented out part relevant to the implementation of `Argmax`?

Reply:
Commented part is implementation of Argmax, which is currently in progress. For now only this line: `* max=func_ptr(1)` has been added to check if the function is passing to the kernel or not.